core/src/test/scala/quasar/destination/gbq/GBQDestinationSpec.scala

```diff
@@ -0,0 +1,286 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+//TODO: don't use this anymore
+//import slamdata.Predef._
+
+
+
+
+
+import argonaut._, Argonaut._
+
+import cats.data.NonEmptyList
+import cats.effect.{ConcurrentEffect, Blocker, IO, Resource, Timer}
+import cats.implicits._
+
+import fs2.{Stream, text}
+
+import java.nio.file.{Files, Paths}
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.concurrent.Executors
+import java.util.UUID
+
+import org.http4s.argonaut.jsonEncoderOf
+import org.http4s.client.Client
+import org.http4s.{
+  AuthScheme, 
+  Credentials, 
+  EntityEncoder, 
+  Method, 
+  Request, 
+  Status,
+  Uri}
+import org.http4s.headers.Authorization
+import org.http4s.headers.`Content-Type`
+import org.http4s.client._
+
+import quasar.api.{Column, ColumnType}
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.connector.destination.{Destination, ResultSink}
+import quasar.connector.render.RenderConfig
+import quasar.api.resource.{ResourceName, ResourcePath}
+import quasar.api.destination.{DestinationError, DestinationType}
+import quasar.connector.ResourceError
+import quasar.contrib.scalaz.MonadError_
+import quasar.EffectfulQSpec
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scala.{
+  Either,
+  List,
+  StringContext,
+}
+import scala.concurrent.duration._
+import scala.Predef.String
+import scala.util.Right
+import scala.util.Left
+
+import scalaz.{-\/,\/-}
+
+import slamdata.Predef.RuntimeException
+
+import shims._
+
+
+object GBQDestinationSpec extends EffectfulQSpec[IO] {
+  sequential
+
+  implicit val timer: Timer[IO] = IO.timer(ExecutionContext.global)
+
+  val testProject = "precog-ci-275718"
+  val testDataset = "dataset_" + UUID.randomUUID().toString.replace("-", "_").toString
+  val tableName = "table_" + UUID.randomUUID().toString.replace("-", "_").toString
+  val authCfgPath = Paths.get(getClass.getClassLoader.getResource("precog-ci-275718-e913743ebfeb.json").toURI)
+  val authCfgString = new String(Files.readAllBytes(authCfgPath), UTF_8)
+  val authCfgJson: Json = Parse.parse(authCfgString) match {
+    case Left(value) => Json.obj("malformed" := true)
+    case Right(value) => value
+  }
+  val gbqCfg = config(authCfgJson, testDataset)
+  val blockingPool = Executors.newFixedThreadPool(5)
+  val blocker = Blocker.liftExecutorService(blockingPool)
+  val httpClient: Client[IO] = JavaNetClientBuilder[IO](blocker).create
+
+  "csv link" should {
+    "reject empty paths with NotAResource" >>* {
```

Comment:
So, we can't check if path is not a resource w/o instantiating destination, can we?